src/compiler/checker.ts

```diff
@@ -22619,7 +22619,12 @@ namespace ts {
          * we want to get the iterated type of an iterable for ES2015 or later, or the iterated type
          * of a iterable (if defined globally) or element type of an array like for ES2015 or earlier.
          */
-        function getIteratedTypeOrElementType(inputType: Type, errorNode: Node, allowStringInput: boolean, allowAsyncIterables: boolean, checkAssignability: boolean): Type {
+        function getIteratedTypeOrElementType(inputType: Type, errorNode: Node, allowStringInput: boolean, allowAsyncIterables: boolean, checkAssignability: boolean): Type | undefined {
+            if (inputType === neverType) {
+                reportTypeNotIterableError(errorNode, allowAsyncIterables);

```

Comment:
Hmmm... the diagnostic message `Diagnostics.Type_must_have_a_Symbol_iterator_method_that_returns_an_iterator` doesn't mention what type we're erroring on. Might be an improvement to include the type in the error, especially since typing `x[Symbol.iterator]()` isn't going to produce an error when x is `never` (and will have a "return type" assignable to an iterator).